### ISSUE TITLE
Homestead versions with box 6.0.0 fail to be provisioned with "Remote connection disconnect. Retrying..." repeated until hard-failure on Vagrant > 2.0.3

### ISSUE THREAD

**Issue by cbj4074** [laravel/homestead#862]
## Versions
  - Vagrant: 2.0.4
  - Provider: Virtualbox 5.2.12
  - Homestead: 7.4.2

## Host operating system
Windows 7 x64

## Homestead.yaml
```yaml
---
version: 6.0.0
ip: "192.168.10.10"
memory: 2048
cpus: 1
provider: virtualbox

authorize: ~/.ssh/homestead.pub

keys:
    - ~/.ssh/homestead

folders:
    - map: C:\Users\me\www
      to: /home/vagrant/Code/www

sites:
    - map: web.site
      to: /home/vagrant/Code/www/public

databases:
    - www

mariadb: true
```

## Vagrant destroy & up output
https://gist.github.com/cbj4074/fcc5cee405b0d176b88aac633cf4069b

## Expected behavior
Homestead should be provisioned successfully.

## Actual behavior
Homestead fails to be provisioned after Vagrant inserts new, secure SSH key, with "Remote connection disconnect. Retrying..." repeated until hard-failure. To be clear, I'm not sure that the key replacement is *causing* the failure; it may simply be the last action that occurs successfully.

I've found two workarounds to this problem:

1. Downgrading the Homestead box version, to `5.2.0`; of course, the fundamental difference between Homestead box versions 5.2.0 and 6.0.0 is that that the former runs Ubuntu 16.04 LTS, while the latter runs 18.04 LTS
2. Downgrading Vagrant to 2.0.3

## Steps to reproduce
1. Attempt to provision a new VM with Vagrant `2.0.4`, Homestead box version `6.0.0`, and Homestead version `7.4.2`
2. Observe failure as described above

## References
It seems that this may be related, because as at least one commenter suggests, downgrading to Vagrant 2.0.3 resolves the issue:

https://github.com/hashicorp/vagrant/issues/9428

**Comment by svpernova09** [comment]
Unfortunately this is out of our hands. The issue lies somewhere with Vagrant *on Windows*, which the Vagrant team seems to consistently keep breaking random things.

Good to know downgrading Vagrant is an option, but I'm willing to bet Vagrant pre 2.1.x won't work with the 18.04 series box (v6.0.0 and up) due to the Linux kernel changes.

**Comment by cbj4074** [comment]
Roger that, @svpernova09 . Thanks for taking a look.

Upon further testing, I can confirm that the same problem exists when using Vagrant >= 2.0.4, even with Homestead box version 5.2.0. So, as you suggested, the problem does appear to be with Vagrant (and possibly only on Windows).

Unfortunately, the problem also exists in all Vagrant versions >= 2.0.4 (yes, I took the time to try them all).

The final takeaway: Windows users shouldn't upgrade Vagrant past 2.0.3 until this issue, whatever it may be, is resolved on the Vagrant side, or Homestead will cease to provision properly.

**Comment by svpernova09** [comment]
awesome. thanks for testing all those versions!

**Comment by cbj4074** [comment]
Sure thing!

Interestingly, Homestead box 6.0.0 does work on Vagrant 2.0.3.

The only snag I've encountered is that PHP's OCI8 extension doesn't work (no errors, but it's not loaded), but that's another matter entirely. I need to perform further testing to determine the issue, but it's probably a bug on the PHP side.

**Comment by cbj4074** [comment]
To follow-up, it seems that the problem is fixed in Vagrant 2.1.2.

I just provisioned a fresh Homestead instance (box version v6.1.0), and it works fine on Vagrant 2.1.2.

Also, I tried with Virtualbox 5.2.12 and 5.2.16 (and the latest Guest Additions from each) and it works correctly on both.

So, it appears that Windows users can simply upgrade everything to the latest versions to be back in business.

**Comment by svpernova09** [comment]
@cbj4074 Thanks for the update! Unfortunately the past several Vagrant releases have broken really weird things on Windows. This is seems to be the new normal >.< If I was a much better Windows Dev & had the free time I'd try to pitch in on the Vagrant project, but not in the cards for me.
